wavefront/resource_external_link.go

```diff
@@ -0,0 +1,186 @@
+package wavefront
+
+import (
+	"fmt"
+
+	"github.com/WavefrontHQ/go-wavefront-management-api"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+const (
+	elNameKey                  = "name"
+	elDescriptionKey           = "description"
+	elTemplateKey              = "template"
+	elMetricFilterRegexKey     = "metric_filter_regex"
+	elSourceFilterRegexKey     = "source_filter_regex"
+	elPointTagFilterRegexesKey = "point_tag_filter_regexes"
+	elIsLogIntegrationKey      = "is_log_integration"
+)
+
+func resourceExternalLink() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceExternalLinkCreate,
+		Read:   resourceExternalLinkRead,
+		Update: resourceExternalLinkUpdate,
+		Delete: resourceExternalLinkDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			elNameKey: {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			elDescriptionKey: {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			elTemplateKey: {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			elMetricFilterRegexKey: {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			elSourceFilterRegexKey: {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			elPointTagFilterRegexesKey: {
+				Type:     schema.TypeMap,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
+			},
+			elIsLogIntegrationKey: {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceExternalLinkCreate(
+	d *schema.ResourceData, meta interface{}) error {
+	externalLinks := meta.(*wavefrontClient).client.ExternalLinks()
+
+	externalLink := wavefront.ExternalLink{
+		Name:                  d.Get(elNameKey).(string),
+		Description:           d.Get(elDescriptionKey).(string),
+		Template:              d.Get(elTemplateKey).(string),
+		MetricFilterRegex:     d.Get(elMetricFilterRegexKey).(string),
+		SourceFilterRegex:     d.Get(elSourceFilterRegexKey).(string),
+		PointTagFilterRegexes: getStringMap(d, elPointTagFilterRegexesKey),
+		IsLogIntegration:      d.Get(elIsLogIntegrationKey).(bool),
+	}
+	err := externalLinks.Create(&externalLink)
+	if err != nil {
+		return fmt.Errorf(
+			"failed to create new Wavefront External Link, %s", err)
+	}
+	d.SetId(*externalLink.ID)
+	return resourceExternalLinkRead(d, meta)
+}
+
+func resourceExternalLinkRead(
+	d *schema.ResourceData, meta interface{}) error {
+	externalLinks := meta.(*wavefrontClient).client.ExternalLinks()
+	id := d.Id()
+	el := wavefront.ExternalLink{ID: &id}
+	err := externalLinks.Get(&el)
+	if wavefront.NotFound(err) {
+		d.SetId("")
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf(
+			"error finding Wavefront External Link, %s. %s",
+			d.Id(),
+			err)
+	}
+	if err := d.Set(elNameKey, el.Name); err != nil {
```

Comment:
Interesting...most times I don't see error handling for d.Set() calls. Was this done due to lint?

Reply:
I mostly did it to help me in case I misspell a key or I try to store the wrong go type under a key.